jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt

```diff
@@ -38,7 +41,16 @@ class AwsConnectionSettingsSelector(
     }
 
     private fun fireChange() {
-        settingsChangedListener(view.region.selectedRegion, view.credentialProvider.getSelectedCredentialsProvider())
+        val connectionSettings = view.region.selectedRegion?.let { region ->
+            view.credentialProvider.getSelectedCredentialsProvider()?.let { credId ->
+                val manager = CredentialManager.getInstance()
+                manager.getCredentialIdentifierById(credId)?.let {
```

Comment:
Can we invert some of these `?.let`'s, I think 3 deep is a bit too deep

Reply:
Converted